### PR TITLE
Non-fatal compile_sync failures

### PR DESCRIPTION
**Pull request by afinch7**
Handle compiler worker failures, so the worker can have multiple failures without exiting.

### CODE REVIEW

File: cli/compiler.rs [hunk 1]
```diff
@@ -164,6 +178,39 @@ fn lazy_start(parent_state: Arc<IsolateState>) -> CompilerShared {
               Ok(())
             })
           }));
+          // All worker responses are here initially before being sent via
```
Comment (ry) [inline cli/compiler.rs]: initialized

File: cli/compiler.rs [hunk 2]
```diff
@@ -164,6 +178,39 @@ fn lazy_start(parent_state: Arc<IsolateState>) -> CompilerShared {
...
+          // resolve many futures via the same channel.
+          runtime.spawn(lazy(move || {
+            debug!("Start worker stream handler!");
+            let worker_stream = resources::get_message_stream_from_worker(rid);
+            worker_stream
+              .for_each(|msg: Buf| {
+                let res_json = std::str::from_utf8(&msg).unwrap();
+                debug!("Got message from worker: {}", res_json);
+                // Get the intended receiver from the message.
+                let cmd_id =
```
Comment (ry) [inline cli/compiler.rs]: Can you abstract out the below stanza into a stand alone function? (ideally with a unit test)

File: cli/compiler.rs [hunk 3]
```diff
@@ -133,6 +142,43 @@ impl ModuleMetaData {
...
+  worker_stream
+    .for_each(|msg: Buf| {
+      // All worker responses are handled here first before being sent via
+      // their respective sender. This system can be compared to the
+      // promise system used on the js side. This provides a way to
+      // resolve many futures via the same channel.
+      let res_json = std::str::from_utf8(&msg).unwrap();
+      debug!("Got message from worker: {}", res_json);
+      // Get the intended receiver's cmd_id from the message.
+      let cmd_id = match serde_json::from_str::<serde_json::Value>(res_json) {
```
Comment (ry) [inline cli/compiler.rs]: I meant just the parsing part - seems like a good function.

Comment (afinch7) [reply cli/compiler.rs]: for cmd_id?

**Comment by ry** [comment]
```
tests/error_004_missing_module → tests/error_004_missing_module.test
```
Sorry for not catching this in the last review 🤦‍♂️

**Pull request merged**